lib/Loop.lua

```diff
@@ -222,9 +214,44 @@ function Loop:replaceSystem(old: System, new: System)
 end
 
 local function orderSystemsByDependencies(unscheduledSystems: { System })
-	local sortedUnscheduledSystems = table.clone(unscheduledSystems)
+	local systemPriorityMap = {}
+	local visiting = "v"
+
+	local function systemPriority(system: System)
+		local priority = systemPriorityMap[system]
+
+		if not priority then
+			priority = 0
+
+			systemPriorityMap[system] = visiting
+
+			if type(system) == "table" then
+				if system.after then
+					for _, dependency in system.after do
+						if systemPriorityMap[dependency] ~= visiting then
+							priority = math.max(priority, systemPriority(dependency) + 1)
+						else
+							error(
+								`Unable to get system priority due to cyclic dependency between: \n{systemName(system)} \nAND \n{systemName(
+									dependency
+								)}`
+							)
```

Comment:
I can anticipate this error message being more clear if we indicate this is due to the after field being defined:

```suggestion
							local errorStatement = {
								"Cyclic dependency detected: System '{systemName(system)}' is set to execute after System 'systemName(dependency)', and vice versa. This creates a loop that prevents the systems from being able to execute in a valid order.",
								"To resolve this issue, reconsider the dependencies between these systems. One possible solution is to update the 'after' field from one of the systems.",
							}
							error(table.concat(errorStatement, "\n"))
```